mvc/action-return-types/7.x/WebApiSample/Controllers/ProductsController.ActionResultOfT.cs

```diff
@@ -0,0 +1,43 @@
+namespace WebApiSample.Controllers;
+
+using System.Net.Mime;
+using Microsoft.AspNetCore.Mvc;
+using WebApiSample.Models;
+
+public partial class ProductsController : ControllerBase
+{
+#if ActionResultOfT
```

Comment:
Rather than `#if xyz` I think it would be better to use one of the following approaches to compile all the code

* `[Route("[controller]/[action]")]` and use descriptive action names
* Have `Products2Controller`, Products 3 controller, etc
* `[Route("[controller]/thisRouteName]`

That makes it much easier to test, modify, etc. Otherwise we need a readme explaining how to define the each controller.

You could keep them in multiple files to make them easier to read. For the docs, it's doesn't matter because we use snippets, but code by itself could be easier to read.